python/ray/train/lightgbm/__init__.py

```diff
@@ -1,11 +1,13 @@
 from ray.train.lightgbm._lightgbm_utils import RayTrainReportCallback
+from ray.train.lightgbm.config import LightGBMConfig
 from ray.train.lightgbm.lightgbm_checkpoint import LightGBMCheckpoint
 from ray.train.lightgbm.lightgbm_predictor import LightGBMPredictor
 from ray.train.lightgbm.lightgbm_trainer import LightGBMTrainer
 
 __all__ = [
     "RayTrainReportCallback",
     "LightGBMCheckpoint",
+    "LightGBMConfig",
```

Comment:
nit: Are we exposing this `LightGBMConfig` as a public API?  Seems no special logics in this config, but it's still an argument in the `LightGBMTrainer`.

Reply:
Hmm, yeah there's not really a reason for a user to know about `LightGBMConfig`.

I was trying to keep the API consistent for `LightGBMTrainer` with all the other `DataParallelTrainer`s that take in a backend config. So, `LightGBMConfig` would need to be public for it to be a constructor argument of `v2.LightGBMTrainer`.

Let me remove it from the public `ray.train.lightgbm` namespace for now while `v2.LightGBMTrainer` is still private.